compiler/rustc_parse/src/parser/ty.rs

```diff
@@ -60,6 +60,27 @@ pub(super) enum RecoverReturnSign {
     No,
 }
 
+/// Signals whether to recover from invalid separator in array syntax `[<ty>; <const>]` (the
+/// semicolon)
+///
+/// We don't want this recovery when parsing pre-2018-edition trait method arguments, which don't
+/// allow names or patterns in the arguments. Example:
+///
+/// ```rust
```

Comment:
This is unnecessary; it's the default.

```suggestion
/// ```
```